test/components/download_link_test.rb

```diff
@@ -0,0 +1,29 @@
+require 'component_test_helper'
+
+class DownloadLinkTest < ComponentTestCase
+  def component_name
+    "download-link"
+  end
+
+  test "fails to render a download link when no href is given" do
+    assert_raise do
+      render_component({})
+    end
+  end
+
+  test "fails to render a download link when no href is given but custom text is" do
+    assert_raise do
+      render_component(download_link_text: 'link text')
```

Comment:
Hmm I think we can remove this test since it seems to be redundant since we changed to `link_text` and it didnt fail.